app/web/src/components/FuncEditor/AttributeBindings.vue

```diff
@@ -100,52 +99,65 @@ const props = defineProps<{
 
 const bindingsModalRef = ref<InstanceType<typeof AttributeBindingsModal>>();
 
-const emit = defineEmits<{
-  (e: "update:modelValue", v: AttributeAssociations): void;
-  (e: "change", v: AttributeAssociations): void;
-}>();
-
-const associations = ref(props.modelValue);
-
-watch(
-  () => props.modelValue,
-  (mv) => {
-    associations.value = mv;
-  },
-  { immediate: true },
+const associations = computed(
+  () =>
+    funcStore.funcDetailsById[funcId.value as string]
+      ?.associations as AttributeAssociations,
 );
 
+const funcArguments = computed(() => funcStore.funcArguments);
+const funcId = computed(() => funcStore.selectedFuncId);
+
 const makeEmptyPrototype = (): AttributePrototypeBag => ({
   id: nilId(),
   componentId: nilId(),
   propId: nilId(),
-  prototypeArguments: associations.value.arguments.map(({ id }) => ({
-    funcArgumentId: id,
-  })),
+  prototypeArguments: funcArguments.value
+    ? funcArguments.value.map(({ id }) => ({
+        funcArgumentId: id,
+      }))
+    : [],
+});
+
+// todo: Don't remove existing configured args when pulling in the new ones
```

Comment:
nice to have: right now when we edit a binding, we're wiping any currently configured args. We should instead keep the currently configured ones, letting them configure any new ones if they exist (and change existing if necessary)